src/js/apps/patients/worklist/filters_app.js

```diff
@@ -36,13 +48,31 @@ export default App.extend({
     this.showAllFiltersButtonView();
     this.showGroupsFilterView();
   },
+  getFiltersCount() {
+    const state = this.getState();
+
+    const customFiltersState = omit(state.toJSON(), 'states');
+    const customFiltersSize = size(omit(customFiltersState, function(value) {
+      return isNull(value);
+    }));
+
+    const statesFilterState = state.get('states');
+    const defaultStates = this.getDefaultStates().map('id').reduce((states, id) => {
+      states[id] = true;
+      return states;
+    }, {});
+
+    const statesFilterCount = !isEqual(defaultStates, statesFilterState) ? 1 : 0;
+
+    return customFiltersSize + statesFilterCount;
+  },
```

Comment:
This calculates a total filters count. 

Number of custom filters + whether or not a states filter was selected (either `1` or `0`).

If the states filter is different than the default (user changed it), I count it as `1`. If the states filter has not been changed, I count it as `0`.